### ISSUE TITLE
Sudden Loss Explosion, followed by total missevaluation

### ISSUE THREAD

**Issue by HellMood** [tensorflow/tensorflow#800]
Hello altogether =)

I'm testing Tensorflow 0.60 (GPU Version) for Python 3.4 on Ubuntu 15.10. The used GPU is a GT 960m. Testing the CIFAR10 Demo for over two days in a row gave me the ~86.x% everbody gets after evaluation. After transforming my own data into batches (cifar10 demo like) and adjusting the cifar-python files accordingly, i achieved the following results for my binary classification task (always stopped training, then evaluating, then restarting).
 
after 240 steps (of 128 size batches)  -- >> 92 %
after 420 steps (of 128 size batches)  -- >> 92 %
after 1500 steps (of 128 size batches)  -- >> 94 %
after 2000 steps (of 128 size batches)  -- >> 95 %

Over the course of these tests, the loss value - printed on the console and retrieved via tensorboard, went from about 12 to about 3. Over the (last) weekend, i raised the maximum steps to 10 million, and let things just run. Today in the morning i did an evaluation on the network, which arrived at a loss of 0.6 but evaluated with 50.4 % which looks like totally random (binary task). Scrolling through the logs i realized, there was a sudden loss explosion at around step 70000, jumping from like 2 to 2000 (!!), and - after slowly recovering from that, again up to about 2 000 000 (!!) million, and also slowly recovering from that downto said 0.6, but with the said total missevaluation. Also noteworthy, the speed from almost constantly 100 examples per second broke downto 30 when the loss spikes occured. The eventfile (one of them) went up to 9 Gigabytes in size, and re-checking with tensorboard gave strange results (showing sometimes 4 million, and sometimes 900 million learn steps, appeared to be totally broken somehow).

Appart from just a different batch file, and adjusted image size parameters, the network is exactly the same as in CIFAR 10.

I'm sorry, i can't provide the logs, but right now i am relearning the whole thing and monitoring if this happens again.

Did anybody of experience something like this before?

**Comment by HellMood** [comment]
Providing additional data after relearning to step 11000 : 
![loss plot](http://i.imgur.com/fPFEImG.png)

steps _____eval_________loss
03000_____96.064%_____1.57
04000_____97.389%_____0.77
06850_____97.854%_____0.17
08000_____98.487%_____0.13
09000_____98.467%_____0.07
10000_____98.556%_____0.08
11000_____98.556%_____0.09

so far, it looks quite like expected, i will just let it run over night again

**Comment by vrv** [comment]
(this type of question belongs on StackOverflow, github is for bugs / feature requests, etc).

**Comment by HellMood** [comment]
it happenend again, this time at (around) step 47030. Sudden loss jump from average 0.05 to 1.8 (thank god it wasnt 2 million agin), then slowly back to 0.71 @ step 55000 ( i stopped there). The model seems totally corrupted after that, evaluation gives 48%, which is worse than throwing a dice for a binary task! I understand that i would have to identify the bug for reporting this as a bug, but i since i use the default out of the box model for cifar10, this horrible behaviour doesn't look much like just "bad learning". I will reroll to step 11000 and monitor again. 

![loss plot2](http://i.imgur.com/V8xpR5U.png)

**Comment by twimnox** [comment]
What was your learning rate? What kind of data are you feeding? Right now i'm doing similar changes to cifar10 model to my own 200x200 vegetation images classification. The datasets have some noise (mixed classes).

My model decreases from 11.6 loss to 3.6 loss after 2k, and then jumps to 300 at 3k and stabilizes at ~80.

My ILR is 0.07 and decay is also 0.07

**Comment by HellMood** [comment]
I used the normal learning rates (and parameters) from the standard CIFAR10 models. My way of successfully combating this problem was actually kind of working around it : Since i work with very suspicious and paradox data (difficult and human labeled), sometimes, due to contradicting information, there is a sequence of learning examples generating very high errors. So i simply reduced the batch size to not stack these errors too high, and in case of failure (see above) i simply switched back to an "OK" model before that point.

I'd suggest you check (evaluate) your models once before and then after the loss jump. Chances are high, that the model after the loss jump is unusable.

**Comment by twimnox** [comment]
@HellMood  that is probably my problem. I'm also using difficult and human labeled data which in some cases might be mislabeled. If I decrease my learning rate from 0.07 to 0.02 the training goes on without loss peaks. I'm also using a small batch size (16 to 32). Using a 32 size will most probably induce my loss to sudden rise, agreeing with what you said about reducing the batch size to not stack errors too high.

You're telling me that if however a model loss peaks at K steps, you will use the model at K-n steps (step where the loss is still stably converging)?

**Comment by HellMood** [comment]
Yes, and guided by the evaluation values. If, for a binary classification task, the evaluation drops to "random" after a loss explosion, (eval(k) = 0.50) and i can ensure that before this happened, everything was okay (eval(k-1) = 0.99), then i reroll to model (k-1)

**Comment by niffler92** [comment]
Hi, I'm also having a similar issue. 
I changed my cifar-10 into binary classification with some adjustments, and it doesn't seem to be doing the right feature extraction as training step increases.
For my case, within 100 steps, the predicted class is all set to one label and shows no change after.
As I see my fully connected layer, I can see that the nodes are mostly dead to one value and the ones with distributed values are looking same.
I tried to lower my learning rate but has the same problem.
Any suggestions?
